### PR TITLE
API1000 support for Deployment plan

### PR DESCRIPTION
**Pull request by AnushaY1916**
### Description
API 1000 Support for Deployment Plan
### Issues Resolved
NA

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass for go 1.11.x + gofmt checks .
- [x] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
  - [X] New endpoints supported are updated in the endpoints-support.md file.
- [X] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,10 +2,14 @@ All notable changes to this project will be documented in this file.
...
 
 # [v1.0.2] (UnReleased)
+
+This version supports OneView appliances with version 4.2 using the OneView REST API version 1000.
+
 - Support for Update Compliance in Logical Interconnect.
 - Usecase for synergy infrastructure provisioning with Network.
 
 ### Oneview Features supported
+- Deployment Plan
```
Comment (madhav-bharadwaj) [inline CHANGELOG.md]: Same suggestion as in the GO PR. This is not a Oneview feature.

File: endpoints-support.md [hunk 2]
```diff
@@ -98,9 +98,13 @@
...
 |     **OS Deployment Plans**                                                                                                                           |
 |<sub>/rest/os-deployment-plans</sub>                                                         | POST     | :white_check_mark:   | :white_check_mark:   | :heavy_multiplication_x:   | :heavy_multiplication_x:   | :heavy_multiplication_x:   |
 |<sub>/rest/os-deployment-plans</sub>                                                         | DELETE   | :white_check_mark:   | :white_check_mark:   | :heavy_multiplication_x:   | :heavy_multiplication_x:   | :heavy_multiplication_x:   |
 <sub>/rest/os-deployment-plans</sub>                                                          | PUT      | :white_check_mark:   | :white_check_mark:   | :heavy_multiplication_x:   | :heavy_multiplication_x:   | :heavy_multiplication_x:   |
+|     **Deployment Plans**                                                                                                                             |
```
Comment (madhav-bharadwaj) [inline endpoints-support.md]: Do we add the `get` endpoint as well here, as we have a datasource?

File: examples/deployent_plan.tf [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+	{
+		constraints = "{}"
+		editable = true
+		id = "1075adb8-5399-41fe-b6b3-71bea8836615"
+		name = "User1Name"
+		type = "string"
+		visible = true
+	},
+	]
+}*/
```
Comment (madhav-bharadwaj) [inline examples/deployent_plan.tf]: `*/` can be removed i guess.

Comment (AnushaY1916) [reply examples/deployent_plan.tf]: Removed

File: endpoints-support.md (hunk 2 above)
Comment (AnushaY1916) [reply endpoints-support.md]: Added

File: oneview/resource_deployment_plan.go [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+				Required: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "OEDeploymentPlanV5",
```
Comment (madhav-bharadwaj) [inline oneview/resource_deployment_plan.go]: My suggestion would be to not include any defaults as in future for higher versions, we cannot change it as it would be a breaking change.

File: CHANGELOG.md (hunk 1 above)
Comment (AnushaY1916) [reply CHANGELOG.md]: Changed

File: oneview/resource_storage_system.go [hunk 5]
```diff
@@ -185,7 +185,7 @@ func resourceStorageSystem() *schema.Resource {
 func resourceStorageSystemCreate(d *schema.ResourceData, meta interface{}) error {
```
Comment (madhav-bharadwaj) [inline oneview/resource_storage_system.go]: These files do not belong to this PR. Can you please check?

File: oneview/resource_storage_system_test.go [hunk 6]
```diff
@@ -20,7 +20,7 @@ import (
 )
```
Comment (madhav-bharadwaj) [inline oneview/resource_storage_system_test.go]: same as above

File: oneview/resource_deployment_plan.go (hunk 4 above)
Comment (AnushaY1916) [reply oneview/resource_deployment_plan.go]: okay. Since the value was same from 500 version, I kept it as default. Anyhow I will remove it.

File: oneview/resource_storage_system.go (hunk 5 above)
Comment (AnushaY1916) [reply oneview/resource_storage_system.go]: StorageSystem Struct got renamed, when updated the govendor dependencies, I got these compilation errors so fixed these.

File: oneview/resource_storage_system_test.go (hunk 6 above)
Comment (AnushaY1916) [reply oneview/resource_storage_system_test.go]: StorageSystem Struct got renamed, when updated the govendor dependencies, I got these compilation errors so fixed these.

File: oneview/resource_storage_system.go (hunk 5 above)
Comment (madhav-bharadwaj) [reply oneview/resource_storage_system.go]: you can just checkout the vendor files for storage_system resource as it is not required for this PR. They can be added in the PR for storage system resource.
